### DISCUSSION TITLE
How to insert entry from history but don't execute

### DISCUSSION THREAD

**Discussion by weilbith** [marlonrichert/zsh-autocomplete#356] [Q&A] [answered]
Just the title. But in fact to then edit the selected command.

**Comment by marlonrichert** [comment, marked as answer]
Just start editing the line. 🙂

**Reply by weilbith** [reply]
Ah okay, I think my issue is that I can't `backspace`. It works like the arrow key up.

**Reply by weilbith** [reply]
And usually when I select an entry with `return` it replaces the last char with a `;` which makes many commands fail.

**Reply by marlonrichert** [reply]
> Ah okay, I think my issue is that I can't `backspace`. It works like the arrow key up.

Is that a problem with your terminal or perhaps something you've set yourself in your dotfiles somewhere? It's not something that `zsh-autocomplete` configures.

> And usually when I select an entry with `return` it replaces the last char with a `;` which makes many commands fail.

It's not supposed to _replace_ the last char with a `;`, it's supposed to only _add_ a `:`. Can you make me a test case with which I can reproduce this? Although, this, too, could be problem with your terminal. Which terminal do you use and what is the value of `$TERM`?

**Reply by weilbith** [reply]
> Is that a problem with your terminal or perhaps something you've set yourself in your dotfiles somewhere? It's not something that zsh-autocomplete configures.

I have this mapping in my `.zshrc` for `backspace`: `bindkey -M viins '^?' backward-delete-char` 

> It's not supposed to replace the last char with a ;, it's supposed to only add a :. Can you make me a test case with which I can reproduce this? Although, this, too, could be problem with your terminal. Which terminal do you use and what is the value of $TERM?

I don't really know how I should provide such an example. It is always the case for me. 🤷🏾 
I use _Alacritty_ and my `$TERM` is `xterm-256color`.

**Reply by weilbith** [reply]
Okay so for example I have the following entry in my history: `bash ./math-script.sh add 1 2`. When I recall this from the history and hit enter it executes `bash ./math-script add 1 ;` which will cause an error.

**Reply by marlonrichert** [reply]
What happens if you recall a line from history and then press Space, instead of Enter?

**Reply by marlonrichert** [reply]
Can you try the following?

1. Run `bindkey '^V' describe-key-briefly`.
2. Press <kbd>Control</kbd> <kbd>V</kbd>, then  <kbd>Backspace</kbd>.

What output does that give?

**Reply by weilbith** [reply]
> What happens if you recall a line from history and then press Space, instead of Enter?

It just adds a space. But I think I have issues to reproduce it 100%. It did happen multiple times today again, but right now it doesn't. 🤔

**Reply by weilbith** [reply]
> Can you try the following?
> 
> 1. Run `bindkey '^V' describe-key-briefly`.
> 2. Press <kbd>Control</kbd> <kbd>V</kbd>, then  <kbd>Backspace</kbd>.
> 
> What output does that give?

Ah, interesting: `"^?" is autopair-delete`

**Reply by marlonrichert** [reply]
> > What happens if you recall a line from history and then press Space, instead of Enter?
> 
> It just adds a space.

Well, yes, of course it does. But what happens now to the character to the left of the `;` that is inserted? Does it remain intact? Or does it disappear like with Enter?

**Reply by marlonrichert** [reply]
> Ah, interesting: `"^?" is autopair-delete`

That's likely the culprit for your unusual Backspace behavior, then. Try if this fixes it:
```zsh
bindkey -M menuselect '^?' .autopair-delete
```
If it doesn't, then try this:
```zsh
bindkey -M menuselect '^?' .backward-delete-char
```

**Reply by marlonrichert** [reply]
What output do you from running `describe-key-briefly` on your Enter key?
